sound/soc/sof/intel/hda.c

```diff
@@ -1057,8 +1116,30 @@ static int hda_sdw_machine_select(struct snd_sof_dev *sdev)
 	 */
 	if (link_mask && !pdata->machine) {
 		mach = pdata->desc->alt_machines;
-		while (mach && mach->link_mask && mach->link_mask != link_mask)
-			mach++;
+		while (mach && mach->link_mask) {
+			if (mach->link_mask != link_mask) {
+				mach++;
+				continue;
+			}
+			/* No need to match adr if there is no links defined */
+			if (!mach->links)
+				break;
+
+			link = mach->links;
+			for (i = 0; i < hdev->info.count; i++, link++) {
+				/*
+				 * Try next machine if any expected Slaves
```

Comment:
couls you please make the S in slaves small case?

Reply:
Updated, thanks.

Reply:
no, we use Slave and Master to refer to the MIPI spec where they are also capitalized.